source/material_model/visco_plastic.cc

```diff
@@ -173,6 +173,22 @@ namespace aspect
                                                   (constants::gas_constant*temperature*stress_exponents_dislocation[j])) *
                                          std::pow(edot_ii,((1. - stress_exponents_dislocation[j])/stress_exponents_dislocation[j]));
 
+          // Allow for different flow-law parameters (grain-size and activation volume) in the lower mantle
+          if (pressure > pressure_lower_mantle)
```

Comment:
Quick question/comment before a full review. How about moving this conditional statement to the beginning of the loop through each compositional field (e.g., right below explanation of viscosity equation)? 

Rather than repeating the equations, `if (pressure > pressure_lower_mantle)` would then define whether the activation volume and grain size would be the 'normal' values or those for the lower mantle. The new variable names would be 'activation_volume' and something like `current_grain_size`?

Reply:
I'm fine with this suggestion - I had tried to make the modification with the smallest possible change to the current code. 

Is there a guiding principal to use in making this decision... when is it better to introduce 2 new variables that are standing in for something else?  Maybe clarity and/or efficiency?

Here it seems like it might be okay, because you define them and immediately use them, so you don't have to search through the code to see where something got defined. This would make the code clearer (I think).  I can't use "activation_volume" because there will still be two - for diffusion and dislocation creep, but maybe current_activation_volume_diffusion and current_activation_volume_dislocation?